haystack/nodes/other/shaper.py

```diff
@@ -0,0 +1,408 @@
+from typing import Optional, List, Dict, Any, Tuple, Union, Callable
+
+import logging
+
+from haystack.nodes.base import BaseComponent
+from haystack.schema import Document, Answer, MultiLabel
+
+
+logger = logging.getLogger(__name__)
+
+
+def rename(value: Any) -> Tuple[Any]:
+    """
+    Identity function. Can be used to rename values in the invocation context without changing them.
+
+    Example:
+
+    ```python
+    assert rename(1) == (1, )
+    ```
+    """
+    return (value,)
+
+
+def value_to_list(value: Any, target_list: List[Any]) -> Tuple[List[Any]]:
+    """
+    Transforms a value into a list containing this value as many times as the length of the target list.
+
+    Example:
+
+    ```python
+    assert value_to_list(value=1, target_list=list(range(5))) == ([1, 1, 1, 1, 1], )
+    ```
+    """
+    return ([value] * len(target_list),)
+
+
+def join_lists(lists: List[List[Any]]) -> Tuple[List[Any]]:
+    """
+    Joins the passed lists into a single one.
+
+    Example:
+
+    ```python
+    assert join_lists(lists=[[1, 2, 3], [4, 5]]) == ([1, 2, 3, 4, 5], )
+    ```
+    """
+    merged_list = []
+    for inner_list in lists:
+        merged_list += inner_list
+    return (merged_list,)
+
+
+def join_strings(strings: List[str], delimiter: str = " ") -> Tuple[str]:
+    """
+    Transforms a list of strings into a single string. The content of this string
+    is the content of all original strings separated by the delimiter you specify.
+
+    Example:
+
+    ```python
+    assert join_strings(strings=["first", "second", "third"], delimiter=" - ") == ("first - second - third", )
+    ```
+    """
+    return (delimiter.join(strings),)
+
+
+def join_documents(documents: List[Document], delimiter: str = " ") -> Tuple[List[Document]]:
+    """
+    Transforms a list of documents into a list containing a single Document. The content of this list
+    is the content of all original documents separated by the delimiter you specify.
+
+    All metadata is dropped. (TODO: fix)
+
+    Example:
+
+    ```python
+    assert join_documents(
+        documents=[
+            Document(content="first"),
+            Document(content="second"),
+            Document(content="third")
+        ],
+        delimiter=" - "
+    ) == ([Document(content="first - second - third")], )
+    ```
+    """
+    return ([Document(content=delimiter.join([d.content for d in documents]))],)
+
+
+def strings_to_answers(strings: List[str]) -> Tuple[List[Answer]]:
+    """
+    Transforms a list of strings into a list of Answers.
+
+    Example:
+
+    ```python
+    assert strings_to_answers(strings=["first", "second", "third"]) == ([
+            Answer(answer="first"),
+            Answer(answer="second"),
+            Answer(answer="third"),
+        ], )
+    ```
+    """
+    return ([Answer(answer=string) for string in strings],)
```

Comment:
> Can we please set type="generative"
 
:+1:

>  And do you see any chance to add the document_ids that have been used as input to the promptnode to the meta

Is this feature critical? Otherwise I'll keep it for the next PR, or this one will never get merged :smile: Features can always be added later.

Reply:
@ZanSara I understand that you want to get this merged and keep the scope small. However, it is an important feature to us. It also seems like this would be a very minor change in the `join_documents` function (this is what you meant right @tstadel ?). Since we already released 1.13 and there is no rush that we have to ship 1.13.1 as fast as possible, I'd vote to add this small change.

Reply:
Unfortunately this is not a small change testing-wise, maybe not even code-wise (it's not immediately evident how to do this without affecting `Shaper`) and will surely delay 1.13.1, which is scheduled for tomorrow. Are you really sure you can't wait until 1.14, which is scheduled for mid February?

Reply:
Let's ship and think about the extension later. As a workaround we can simply take everything from the `documents` key.

Reply:
>  there is no rush that we have to ship 1.13.1 as fast as possible

Not really - Haystack releases take time and I would honestly prefer to wrap up 1.13.1 sooner than later. Additionally,, we're already stretching the "bugfix release" concept by a lot, there are 6 PRs assigned to the .1 milestone and if we keep the window open more requests will come our way. 1.14 is scheduled for February 20, unless a fix is needed to address a high severity bug we should wait.